documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc

```diff
@@ -703,3 +703,40 @@ public interface MovieMapper {
 ====
 Although the used mechanism is the same, the user has to be a bit more careful. Refactoring the name of a defined qualifier in an IDE will neatly refactor all other occurrences as well. This is obviously not the case for changing a name.
 ====
+
+=== Combining qualifiers with defaults
+When using `defaultValue` with `qualifiedByName` or `qualifiedBy` be aware that the default value will the passed into the conversion method. In the case that this is not wanted you need to use `defaultExpression` instead of `defaultValue`.
```

Comment:
Perhaps: a better approach:

Please note that the `Mapping#defaultValue` is in essence a `String`. This `String` needs to be converted to the `Mapping#target` . Providing a  `qualifiedByName` or `qualifiedBy` can lead to errors if the `Mapping#source` is not a `String`, since it forces MapStruct to use that method. To solve this, please provide an approriate mapping method, that transforms the `String` into the desired type of `Mapping#target` and also annotate this method with the same  `qualifiedByName` or `qualifiedBy`  annotation.

I don't think `defaultExprerssion` is intended as an alternative to `defaultValue`.  (do not want to encourage its use).

Reply:
This is a matter of taste: but I really dislike the `defaultExpression`. It was always intended as a kind of a 'gap stop'

Reply:
Or perhaps we should document both. I don't know. If you guys really feel that `defaultExpression` is the way to go.

Reply:
I'm thinking of adjusting the text to the following:

----
Please note that the `Mapping#defaultValue` is in essence a `String`, which needs to be converted to the `Mapping#target`. Providing a `Mapping#qualifiedByName` or `Mapping#qualifiedBy` will force MapStruct to use that method. If you want different behavior for the `Mapping#defaultValue`, then please provide an appropriate mapping method. This mapping method needs to transforms a `String` into the desired type of `Mapping#target` and also be annotated so that it can be found by the `Mapping#qualifiedByName` or `Mapping#qualifiedBy`.
----

This is not yet done in code, but I figured we could discuss it here beforehand. That way we can get a correct description out there for the guide.

I guess the other open bit for this issue is the javadoc enhancements to make it more clear that the `qualifiedByName` and `qualifiedBy` will also be used for the `defaultValue`. I think it would suffice to add this to the javadoc of the `qualifiedBy`/`qualifiedByName` bit, but not needed for the `defaultValue` one.

Reply:
I too am not a big fan of the expression semantics as well. However, I do think that in this case it is OK to mention it. Your last proposal is OK @Zegveld. I would only add one more sentence in the end "An alternative to avoid the usage of the qualified mapping method `defaultValueExpression` can be used.".

>I guess the other open bit for this issue is the javadoc enhancements to make it more clear that the `qualifiedByName` and `qualifiedBy` will also be used for the `defaultValue`. I think it would suffice to add this to the javadoc of the `qualifiedBy`/`qualifiedByName` bit, but not needed for the `defaultValue` one.

I agree that adding extra javadoc to the `qualifiedBy` / `qualifiedByName` to clarify that it is also used for the `defaultValue` is enough.